### PR TITLE
refactor: remove `Dispatcher` and related methods, closes #290

### LINKED ISSUE #290: Remove `Dispatcher` and related methods
`Dispatcher` uses  a `mpsc::channel` to enable sending scripts from any thread to the main thread, where `evaluate_scripts` is supposed to be called. However, using the `Dispatcher` alone does not wake up the event loop, and therefore requires that another event happens before the script can be executed. I believe it would be a lot cleaner if users of `wry` can execute scripts directly (on the main thread) and provide their own mechanism to send scripts to the main thread (e.g. by using `EventLoopProxy` and a custom event). I suggest removing `Dispatcher`, `evaluate_script`and `dispatch_script` after adding a way to execute scripts immediately.

Therefore, I proposed adding a `eval` method in #289. I also created a draft PR for tauri (tauri-apps/tauri#1965) to switch to the new `eval` method and will create a draft PR here to remove the old `Dispatcher` api. Both would remain drafts until a new version of `wry` containing #289 is released.

### PR DESCRIPTION
**Pull request by geieredgar**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [x] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [x] Yes. Issue #290
- [ ] No

**The PR fulfills these requirements:**

- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**
~~Blocked until a new version containing #289 is released.~~

### CODE REVIEW

File: src/webview/mod.rs [hunk 1]
```diff
@@ -325,33 +299,18 @@ impl WebView {
...
-    while let Ok(js) = self.rx.try_recv() {
-      self.webview.eval(&js)?;
-    }
-
-    Ok(())
+  /// Evaluate and run javascript code. Must be called on the same thread who created the
+  /// [`WebView`]. Use [`EventLoopProxy`] and a custom event to send scripts from other threads.
+  ///
+  /// [`EventLoopProxy`]: crate::application::event_loop::EventLoopProxy
+  pub fn eval(&self, js: &str) -> Result<()> {
```
Comment (wusyong) [inline src/webview/mod.rs]: The name of `eval` is more because of historical reasons.
Could you name it `evaluate_script`?

Comment (geieredgar) [reply src/webview/mod.rs]: OK. I closed #289 in favor of this PR, made the necessary adjustments in my draft PR [tauri-apps/tauri#1965](https://github.com/tauri-apps/tauri/pull/1965) and removed the draft status from this PR.

**Review by wusyong — DISMISSED** [review]

**Comment by nothingismagick** [comment]
Awesome! Can you just run `cargo +nightly format`, then this should be good to go.

**Comment by geieredgar** [comment]
@nothingismagick Running `cargo +nightly fmt` did not change anything, so I went ahead and fixed the clippy warning manually in the last added commit. However, the warning is not at all related to my changes. I think it is triggered by a new clippy version in the CI.

**Review by wusyong — APPROVED** [review]
I believe it's because of new clippy lint. I could merge it and fix those later.

**Pull request merged**